src/kibana/plugins/discover/components/field_chooser/field_chooser.js

```diff
@@ -21,10 +21,16 @@ define(function (require) {
         data: '=',
         state: '=',
         indexPattern: '=',
+        indexPatternList: '=',
         updateFilterInQuery: '=filter'
       },
       template: html,
-      controller: function ($scope) {
+      controller: function ($scope, $route) {
+        $scope.setIndexPattern = function (indexPattern) {
+          $scope.state.index = indexPattern;
```

Comment:
This is only 1 of 2 directives where the state is passed in, the other bring the filter bar. There, it makes sense as you don't want to have to re-render the whole app to change the filters. Here, the only time you use state is when you re-rendering anyway, so I think it makes sense not to pass it in.